pkg/manifests/manifests.go

```diff
@@ -883,6 +883,18 @@ func (f *Factory) updateNodeExporterArgs(args []string) []string {
 		args = setArg(args, "--no-collector.ksmd", "")
 	}
 
+	if f.config.ClusterMonitoringConfiguration.NodeExporterConfig.Collectors.Systemd.Enabled {
+		args = setArg(args, "--collector.systemd", "")
+		units := f.config.ClusterMonitoringConfiguration.NodeExporterConfig.Collectors.Systemd.Units
+		for idx, unit := range units {
+			units[idx] = fmt.Sprintf("(%s)", unit)
```

Comment:
we need to use regexp.Compile() to ensure that the input parameters are valid regexps (e.g. similar to https://github.com/openshift/cluster-monitoring-operator/pull/1963)